File: day02/Solution.scala
package day02

import scala.io.Source

case object Solution extends App {

  def partOne(input: String): Int = {
    val diffs = for {
      line <- input.split("\n")
      numbers = line.split("\t").map(_.toInt)
    } yield numbers.max - numbers.min

    diffs.sum
  }

  def partTwo(input: String): Int = {
    val divisions = for {
      line <- input.split("\n")
      numbers = line.split("\t").map(_.toInt)
      a <- numbers
      b <- numbers
      if a > b && a % b == 0
    } yield a / b

    divisions.sum
  }

  val input = Source.fromFile(getClass.getPackage.getName + "/input.in").mkString

  println(partOne(input))
  println(partTwo(input))

}

File: Day02/Solution.cs
using System;
using System.Linq;

namespace AdventOfCode2017.Day02 {

    class Solution : Solver {

        public void Solve(string input) {
            Console.WriteLine(PartOne(input));
            Console.WriteLine(PartTwo(input));
        }

        int PartOne(string input) {
            return (
                from line in input.Split('\n')
                let numbers = line.Split('\t').Select(int.Parse)
                select numbers.Max() - numbers.Min()
            ).Sum();
        }

        int PartTwo(string input) {
            return (
                from line in input.Split('\n')
                let numbers = line.Split('\t').Select(int.Parse)
                from a in numbers
                from b in numbers
                where a > b && a % b == 0
                select a / b
            ).Sum();
        }
    }
}
